### PR TITLE
FIRECERT-2169: [FCS] Implement Setter/Getter validation Glue Code

### PR DESCRIPTION
**Pull request by Abhishk123**

### CODE REVIEW

File: cypress/support/step_definitions/validations.js [hunk 1]
```diff
@@ -390,3 +390,213 @@ Given(
     });
   }
 );
+
+/**
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/validations.js:394]: https://ccp.sys.comcast.net/browse/FIRECERT-2044 will these changes work with this glue code

File: cypress/support/step_definitions/validations.js [hunk 2]
```diff
@@ -390,3 +390,213 @@ Given(
...
+      const context = {};
+      const expectingError = errorContent ? true : false;
+
+      // When fireboltCall object key passed fetching the object from the fireboltCalls data else reading it from environment variable
+      if (fireboltCallKey) {
+        cy.getFireboltData(fireboltCallKey).then((fireboltData) => {
+          fireboltCallObject = fireboltData;
+        });
+      } else {
+        fireboltCallObject = UTILS.getEnvVariable('runtime').fireboltCall;
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/validations.js:432]: 'set' can be moved to a constant

File: cypress/support/step_definitions/validations.js [hunk 3]
```diff
@@ -390,3 +390,215 @@ Given(
...
+        return input.map((item) => resolveContentObject(item));
+      } else if (typeof input == CONSTANTS.TYPE_OBJECT && input !== null) {
+        for (const key in input) {
+          if (Object.hasOwnProperty.call(input, key)) {
+            input[key] = resolveContentObject(input[key]);
+          }
+        }
+        return input;
+      } else if (typeof input === CONSTANTS.TYPE_FUNCTION) {
+        return input();
```
[the remark this answers is not in the document]
Comment (Abhishk123) [reply cypress/support/step_definitions/validations.js:596]: added a condition to check for nul or undefined

File: cypress/support/step_definitions/validations.js (hunk 2 above)
Comment (Abhishk123) [reply cypress/support/step_definitions/validations.js:432]: addressed the comment

File: cypress/support/step_definitions/validations.js (hunk 1 above)
Comment (Abhishk123) [reply cypress/support/step_definitions/validations.js:394]: added the logic to take default values for path and content

**Review by neeradanelxsi — APPROVED** [review]

**Review by jbigel — APPROVED** [review]

**Review by neeradanelxsi — APPROVED** [review]

**Review by Eswar2103 — APPROVED** [review]

**Pull request merged**